src/cloud/common/CloudAuth/OneGateCloudAuth.rb

```diff
@@ -96,21 +95,25 @@ def do_auth(env, params={})
     private
 
     def encrypt(data, token_password)
-        @cipher.encrypt
-        @cipher.key = token_password
+        cipher = OpenSSL::Cipher.new(CIPHER)
 
-        rc = @cipher.update(data)
-        rc << @cipher.final
+        cipher.encrypt
+        cipher.key = token_password
+
+        rc = cipher.update(data)
+        rc << cipher.final
 
         return rc
     end
 
     def decrypt(token_password, data)
-        @cipher.decrypt
-        @cipher.key = token_password[0..31]
+        cipher = OpenSSL::Cipher.new(CIPHER)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Use of weak cipher algorithm, the selected algorithm is considered weak and may be vulnerable to exploit. Use a stronger algorithm such as AES 256 GCM instead.